File: example/typescript/test/TypesModule/Fixtures/Asserts.ts
namespace TypeModule.Assert {
    export interface ExpectedDateRange {
        from?: string,
        to?: string,
    }
    export function dateRange(given: DateRange, expected: ExpectedDateRange) {
        if (expected.from !== undefined) {
            AssertEquals(dateGetValue(dateRangeGetFrom(given)), expected.from)
        }

        if (expected.to !== undefined) {
            AssertEquals(dateGetValue(dateRangeGetTo(given)), expected.to)
        }
    }
}

File: example/kotlin/src/testFixtures/kotlin/com/some/pkg/typesmodule/fixtures/Asserts.kt
package com.some.pkg.typesmodule.fixtures

import org.assertj.core.api.Assertions.assertThat

import com.some.pkg.typesmodule.api.*

data class ExpectedDateRange(
    var from: String? = null,
    var to: String? = null,
)
fun assertDateRange(given: DateRange, expectedInit: ExpectedDateRange.() -> Unit) {
    val expected = ExpectedDateRange().apply(expectedInit)

    expected.from?.let {
        assertThat(dateGetValue(dateRangeGetFrom(given))).isEqualTo(it)
    }

    expected.to?.let {
        assertThat(dateGetValue(dateRangeGetTo(given))).isEqualTo(it)
    }
}